milton-manager/src/main/kotlin/io/terbium/milton/AlgoliaClient.kt

```diff
@@ -83,6 +83,21 @@ class AlgoliaClient @Inject constructor(
         return outputList
     }
 
+    suspend fun deletePage(url: String) {
+        val browseResults = index.browseObjects(Query(
+                query = url,
+                restrictSearchableAttributes = listOf(Attribute("url")),
+                attributesToRetrieve = listOf(
+                        Attribute("url"),
+                        Attribute("objectID")
+                )
+        ))
+        val objectIds = browseResults.flatMap { it.hits }
+                .filter { it["url"]!!.primitive.content == url }
+                .map { ObjectID(it["objectID"]!!.primitive.content) }
+        index.deleteObjects(objectIds)
```

Comment:
Any idea why the ObjectID's you're setting aren't getting persisted? It would be nice to not have to perform this extra search when removing a URL from Milton.

Reply:
Agreed, but no, I'm really pretty sure I'm using the API correctly... My best guess is that the IDs are too long?